backend/src/api/game/game_state_response.rs

```diff
@@ -26,24 +29,21 @@ pub struct GameResponse {
     history: Vec<(String, String)>,
 }
 
-impl GameResponse {
-    // TODO we don't need to pass everything in once we have a couple of good helpers :)
-    pub fn new_from_state(state: &State) -> Self {
-        let white_player = User::new("1", "white", false).unwrap();
-        let black_player = User::new("2", "black", false).unwrap();
-        Self {
-            game_id: state.game_id,
+impl GameStateResponse {
+    pub async fn new_from(game: &Game, state: &State, pool: &DbPool) -> Result<Self, ServerError> {
```

Comment:
i don't like that we're using both `Game` and `State` here -- the structs have a lot of redundant fields, and it's not clear just from reading this function which value to trust as the source of truth (i.e. is this function only valid after `state`'s been synced to the database? are `game`'s fields even valid when this is called?). to clarify this, since we're just using the UIDs from `Game`, what about changing this to:

```suggestion
    pub async fn new_from(white_uid: &str, black_uid: &str, state: &State, pool: &DbPool) -> Result<Self, ServerError> {
```